mako/codegen.py

```diff
@@ -524,54 +521,50 @@ def write_variable_declares(self, identifiers, toplevel=False, limit=None):
         for ident in to_write:
             if ident in comp_idents:
                 comp = comp_idents[ident]
-                if comp.is_block:
-                    if not comp.is_anonymous:
-                        self.write_def_decl(comp, identifiers)
-                    else:
-                        self.write_inline_def(comp, identifiers, nested=True)
+                if (
+                    comp.is_block
+                    and not comp.is_anonymous
+                    or not comp.is_block
+                    and comp.is_root()
+                ):
```

Comment:
**Michael Bourke** (bourke) wrote:

-1 from me here. I appreciate that it's cleaner in terms of control flow, but it's not as easy to read or understand.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319